conda/resolve.py

```diff
@@ -409,14 +424,33 @@ def build_conflict_map(self, specs, specs_to_add=None, history_specs=None):
             # records_per_name is a completely arbitrary number here.  It is meant to gather more
             # than just one record, to explore the space of dependencies a bit.  Doing all of them
             #    can be an enormous problem, though.  This is hopefully a good compromise.
-            records_per_name = 7
-            g = GeneralGraph([_ for v in records_for_graph.values() for _ in v[:records_per_name]])
+            gg = GeneralGraph(
+                [_ for v in records_for_graph.values() for _ in v])
             spec_order = sorted(sdeps_with_dep.keys(),
-                                key=lambda x: list(g.graph_by_name.keys()).index(x.name))
-            for spec in spec_order:
-                # the DFS approach works well when things are actually in the graph
-                bad_deps.extend(g.depth_first_search_by_name(spec, dep, sdeps[spec]))
+                                key=lambda x: list(gg.graph_by_name.keys()).index(x.name))
 
+            for spec in spec_order:
+                allowed_specs = sdeps[spec]
+                ga = GeneralGraph([_ for allowed_pkgs in allowed_specs.values()
```

Comment:
can we use the graph we already have above, but filter it somehow?  Creation of the graph is pretty expensive.